File: leetcode/ruby/reverseLinkedList.rb
# Reverse a singly linked list.

class ListNode
  attr_accessor :val, :next

  def initialize(val)
    @val = val
    @next = nil
  end
end

def reverse(head)
  return head if head.nil?

  curr = head
  prev = nil
  while (curr)
    nex = curr.next
    curr.next = prev
    prev = curr
    curr = next
  end
  prev
end

File: leetcode/js/reverseLinkedList.js
// Reverse a singly linked list.

function ListNode(val) {
  this.val = val;
  this.next = null;
}

const reverse = head => {
  if (!head) return head;
  let curr = head;
  let output;
  while (curr) {
    let next = curr.next;
    curr.next = output;
    output = curr;
    curr = next;
  }
  return output;
};
